src/StreamrClient.js

```diff
@@ -29,33 +29,18 @@ export default class StreamrClient extends EventEmitter {
             autoConnect: true,
             // Automatically disconnect on last unsubscribe
             autoDisconnect: true,
-            auth: {
-                privateKey: null,
-                provider: null,
-                apiKey: null,
-                username: null,
-                password: null,
-                sessionToken: null,
-            },
+            auth: {},
         }
         this.subsByStream = {}
         this.subById = {}
         this.publishQueue = []
 
         Object.assign(this.options, options || {})
 
-        this.session = new Session(this)
-
-        // Backwards compatibility for the use of apiKey in code other than strict authentication by the Session
-        this.options.apiKey = this.options.auth.apiKey
-
-        // Backwards compatibility for option 'authKey' => 'apiKey'
-        if (this.options.authKey && !this.options.apiKey) {
-            this.options.apiKey = this.options.authKey
-        }
+        this.session = new Session(this, options)
```

Comment:
Why not pass `options.auth` to `Session`? It only reads that part and discards the rest, right?